src/main/java/org/jabref/gui/entryeditor/citationrelationtab/RelatedEntriesComponent.java

```diff
@@ -0,0 +1,237 @@
+    private CheckListView<CitationRelationItem> relatedEntriesListView;
+
+    public RelatedEntriesComponent(BibEntry pivotEntry, RelatedEntriesRepository repository, LibraryTab libraryTab,
+                                   RelatedEntriesComponentConfig config, DialogService dialogService,
+                                   BibDatabaseContext databaseContext, UndoManager undoManager,
+                                   StateManager stateManager, FileUpdateMonitor fileUpdateMonitor,
+                                   PreferencesService prefs) {
+        this.pivotEntry = pivotEntry;
+        this.config = config;
+        this.viewModel = new RelatedEntriesComponentViewModel(pivotEntry, repository, dialogService, databaseContext,
+                undoManager, stateManager, fileUpdateMonitor, prefs);
+        this.libraryTab = libraryTab;
+
+        initialize();
+        if (canLoadEntries()) {
+            viewModel.loadEntries();
+        } else {
+            relatedEntriesListView.setPlaceholder(buildLabel(Localization.lang("The selected entry doesn't have a DOI linked to it. Lookup a DOI and try again.")));
+        }
+
+        viewModel.relatedEntriesResultPropertyProperty().addListener((observable, oldValue, result) -> {
+            if (result.isPending()) {
+                showNodes(progressIndicator, cancelButton);
+                hideNodes(refreshButton, importEntriesButton);
+                relatedEntriesListView.getItems().clear();
+                relatedEntriesListView.setPlaceholder(buildLabel(Localization.lang("Loading...")));
+            } else if (result.isSuccess()) {
+                hideNodes(progressIndicator, cancelButton);
+                showNodes(refreshButton, importEntriesButton);
+
+                List<BibEntry> entries = result.asSuccess().value();
+                if (entries.isEmpty()) {
+                    relatedEntriesListView.setPlaceholder(buildLabel(Localization.lang("No articles found")));
+                } else {
+                    relatedEntriesListView.setItems(
+                            FXCollections.observableArrayList(entries.stream().map(entry ->
+                                    new CitationRelationItem(entry, false)).collect(Collectors.toList()))
+                    );
+                }
+            } else if (result.isFailure()) {
+                hideNodes(progressIndicator, cancelButton);
+                showNodes(refreshButton, importEntriesButton);
+
+                Exception exception = result.asFailure().exception();
+
+                relatedEntriesListView.setPlaceholder(buildLabel(
+                        Localization.lang("Error while fetching citing entries: %0", exception.getMessage())));
+
+                LOGGER.error("Error while fetching entry's citation relations", exception);
+            } else {
+                throw new IllegalStateException("Result is can be in one of the three legal states: PENDING, SUCCESS or FAILURE");
```

Comment:
`IllegalSteException` is too hard, because it is a RuntimeException crashing whole JabRef. Jsut do a `LOGGER.error(...)`.

Reply:
I think we could use pattern matching here now that we use Java 21!